airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/integrations/base/ssh/SshBastionContainer.kt

```diff
@@ -0,0 +1,171 @@
+                .put("tunnel_port", containerAddress.right)
+                .put("tunnel_user", SSH_USER)
+                .put(
+                    "tunnel_user_password",
+                    if (tunnelMethod == SshTunnel.TunnelMethod.SSH_PASSWORD_AUTH) SSH_PASSWORD
+                    else ""
+                )
+                .put(
+                    "ssh_key",
+                    if (tunnelMethod == SshTunnel.TunnelMethod.SSH_KEY_AUTH)
+                        container!!.execInContainer("cat", "var/bastion/id_rsa").stdout
+                    else ""
+                )
+                .build()
+        )
+    }
+
+    @Throws(IOException::class, InterruptedException::class)
+    fun getTunnelConfig(
+        tunnelMethod: SshTunnel.TunnelMethod?,
+        builderWithSchema: ImmutableMap.Builder<Any?, Any?>?,
+        innerAddress: Boolean
+    ): JsonNode? {
+        return Jsons.jsonNode(
+            builderWithSchema!!
+                .put("tunnel_method", getTunnelMethod(tunnelMethod, innerAddress))
+                .build()
+        )
+    }
+
+    fun getBasicDbConfigBuider(db: JdbcDatabaseContainer<*>?): ImmutableMap.Builder<Any?, Any?>? {
+        return getBasicDbConfigBuider(db, db!!.databaseName)
+    }
+
+    fun getBasicDbConfigBuider(
+        db: JdbcDatabaseContainer<*>?,
+        schemas: MutableList<String?>?
+    ): ImmutableMap.Builder<Any?, Any?>? {
+        return getBasicDbConfigBuider(db, db!!.databaseName)!!.put("schemas", schemas)
+    }
+
+    fun getBasicDbConfigBuider(
+        db: JdbcDatabaseContainer<*>?,
+        schemaName: String?
+    ): ImmutableMap.Builder<Any?, Any?>? {
+        return ImmutableMap.builder<Any?, Any?>()
+            .put("host", Objects.requireNonNull(HostPortResolver.resolveHost(db)))
+            .put("username", db!!.username)
+            .put("password", db.password)
+            .put("port", HostPortResolver.resolvePort(db))
+            .put("database", schemaName)
+            .put("ssl", false)
+    }
+
+    fun stopAndCloseContainers(db: JdbcDatabaseContainer<*>?) {
+        container!!.stop()
+        container!!.close()
+        db!!.stop()
+        db.close()
+    }
+
+    fun stopAndClose() {
+        container!!.close()
+    }
+
+    override fun close() {
+        stopAndClose()
+    }
+
+    companion object {
+        private val factory: SshBastionContainerFactory? = SshBastionContainerFactory()
+
+        private val SSH_USER: String? = "sshuser"
+        private val SSH_PASSWORD: String? = "secret"
+
+        /**
+         * Returns the inner docker network ip address and port of a container. This can be used to
+         * reach a container from another container running on the same network
+         *
+         * @param container container
+         * @return a pair of host and port
+         */
+        fun getInnerContainerAddress(container: Container<*>?): ImmutablePair<String?, Int?>? {
+            return ImmutablePair.of(
```

Comment:
Kotlin has a built-in [Pair type](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/)